client/app/domain/options/delete/domain-options-delete.controller.js

```diff
@@ -0,0 +1,44 @@
+angular.module('controllers').controller(
+  'controllers.Domain.Options.Delete',
+  class DomainDnsLockCtrl {
+    constructor($scope, $rootScope, $translate, Alerter, Domain) {
+      this.$scope = $scope;
+      this.$rootScope = $rootScope;
+      this.$translate = $translate;
+      this.Alerter = Alerter;
+      this.Domain = Domain;
+    }
+
+    $onInit() {
+      this.loading = false;
+      this.domain = this.$scope.currentActionData.domain;
+      this.option = this.$scope.currentActionData.option;
+      this.$scope.deleteDomain = () => this.deleteDomain();
+    }
+
+    deleteDomain() {
+      this.loading = true;
+      return this.Domain.deleteOption(
+        this.domain.name,
+        this.option.option,
+      )
+        .then(() => this.Alerter.success(
+          this.$translate.instant('domain_tab_options_delete_success'),
+          this.$scope.alerts.main,
+        ))
+        .catch((err) => {
+          _.set(err, 'type', err.type || 'ERROR');
+          this.Alerter.alertFromSWS(
+            this.$translate.instant('domain_tab_options_delete_error'),
+            err,
+            this.$scope.alerts.main,
+          );
+        })
+        .finally(() => {
+          this.loading = false;
+          this.$rootScope.$broadcast('Domain.Options.Delete');
```

Comment:
You should avoid using `$rootScope`. `$scope.$emit` is sufficient in this case :)

Here, by doing $rootScope.$broadcast, you are going to broadcast the event to all scopes in the application.

By doing an `$emit`, you are going to dispatch the event only to the parent scopes :)

Reply:
Understood @jleveugle . But in this case, we will need to catch the `Domain.Options.Delete` event in domain-general-informations.controller.js, which is not the parent scope. Hence, $emit is not going to work in this case. Please let me know if I understood it wrong :)